crates/transaction-pool/src/validate/eth.rs

```diff
@@ -442,8 +464,31 @@ where
             )
         }
 
-        // Checks for max cost
+        #[cfg(not(feature = "optimism"))]
         let cost = transaction.cost();
+
+        #[cfg(feature = "optimism")]
+        let cost = {
+            let block = match self.client.block_by_number_or_tag(BlockNumberOrTag::Latest) {
+                Ok(block) => block.unwrap_or_default(),
```

Comment:
non-existing latest block should be an error imo. we should def avoid using `SealedBlock::default()` since it's meaningless